### DISCUSSION TITLE
How to configure __VUE_OPTIONS_API__ in quasar.conf.js?

### DISCUSSION THREAD

**Discussion by iliubinskii** [quasarframework/quasar#12654] [General - Components / Directives / etc]
\_\_VUE_OPTIONS_API\_\_ is described here:
https://github.com/vuejs/core/blob/main/packages/vue/README.md#bundler-build-feature-flags

The way recommended by Vue is using DefinePlugin:
https://webpack.js.org/plugins/define-plugin/

However, adding:
```ts
const webpack = require("webpack");

new webpack.DefinePlugin({
  __VUE_OPTIONS_API__: JSON.stringify(false | true)
});
```
to quasar.conf.js does not change anything. \_\_VUE_OPTIONS_API\_\_ remains true regardless of what I set using DefinePlugin.

**Comment by MilosPaunovic** [comment]
Have you tried to extend `build` property of your `quasar.conf.js`, like this:
```js
build: {    
  chainWebpack(chain) {
    chain
      .plugin('feature-flags')
      .use(require('webpack').DefinePlugin, [{
        __VUE_OPTIONS_API__: false,
      }])
  }
}
```

**Comment by iliubinskii** [comment]
I tried your code.
It does not change \_\_VUE_OPTIONS_API\_\_ to false and I get this warning:
![image](https://user-images.githubusercontent.com/70687432/156857114-f408bba9-fc81-4dac-843a-4ef6d490a820.png)

I also tried this:
```ts
      extendWebpack(config) {
        const webpack = require("webpack");

        config.plugins = [
          new webpack.DefinePlugin({
            __VUE_OPTIONS_API__: JSON.stringify(false)
          }),
          ...config.plugins
        ];
      },
```

It did change \_\_VUE_OPTIONS_API\_\_ to false but I still get warning.

I think the problem is that Quasar is already using DefinePlugin.
Probably \_\_VUE_OPTIONS_API\_\_  should be configured via Quasar.

**Comment by MilosPaunovic** [comment]
Yeah, Quasar is setting it here, without posibility to explicitly change value:
[https://github.com/quasarframework/quasar/blob/34477a9cefd2f67688b156ec47a62e52f0ed1b64/app-webpack/lib/quasar-conf-file.js#L317](https://github.com/quasarframework/quasar/blob/34477a9cefd2f67688b156ec47a62e52f0ed1b64/app-webpack/lib/quasar-conf-file.js#L317)

So that's probably why you're getting the warning.
